raiden/smart_contracts/ChannelManagerLibrary.sol

```diff
@@ -58,7 +58,12 @@ library ChannelManagerLibrary {
 
         address[] storage existingChannels = self.nodeChannels[msg.sender];
```

Comment:
the question is if we are iterating over an array while at the same time modifying it or if one of the loops works with a copy of the array. 

@LefterisJP Can you please take a look at this issue?